ui-manchette/src/types.ts

```diff
@@ -10,7 +10,13 @@ export type Waypoint = {
 export type InteractiveWaypoint = Waypoint & {
   styles?: CSSProperties;
   display?: boolean;
-  onClick?: (opId: string, opRef: HTMLDivElement | null) => void;
+  onClick?: (waypointId: string, waypointRef: React.RefObject<HTMLDivElement>) => void;
+};
+
+export type WaypointMenuData = {
+  menu: React.ReactNode;
+  activeWaypointId?: string;
+  isWaypointInView?: boolean;
```

Comment:
Instead of adding a new `isWaypointInView` field, can't the caller pass a null `menu` field instead? (I believe `React.ReactNode` allows for `null`.)